docs/upgrade.html

```diff
@@ -68,6 +68,17 @@ <h4><a id="upgrade_2_4_0" href="#upgrade_2_4_0">Upgrading from 0.8.x, 0.9.x, 0.1
     </li>
 </ol>
 
+<p><b>Additional Upgrade Notes:</b></p>
+
+<ol>
+    <li>ZooKeeper has been upgraded to 3.5.6. If the ZooKeeper upgrade from 3.4.X to 3.5.6 is failing
+        with "java.io.IOException: No snapshot found, but there are log entries" error, then we may need to add
+        <a href="https://zookeeper.apache.org/doc/r3.5.6/zookeeperAdmin.html#sc_advancedConfiguration">snapshot.trust.empty=true</a>
+        config to the <code>config/zookeeper.properties</code>. For more details about the issue please refer to
+        <a href="https://issues.apache.org/jira/browse/ZOOKEEPER-3056">ZOOKEEPER-3056</a>.
+    </li>
+</ol>
```

Comment:
We should be prescriptive about it, I think it will always fail if you don't do it.

Reply:
@ijuma Thanks for the review. Upgrade fails only if there are no snapshot files.

Reply:
@omkreddy I think this will always fail if you're upgrading from 3.4 and evidence from people who tried indicate it's true.

Reply:
"We have not run an ensemble without some form of ZOOKEEPER-2325 in years, as such we always have snapshots available and ZooKeeper being unable to load a valid snapshot is a sign that something is very wrong."

https://issues.apache.org/jira/browse/ZOOKEEPER-2325 is not available in the 3.4.x series.

Reply:
@omkreddy Since this is not a troubleshooting guide, we should just say something like:

> ZooKeeper has been upgraded to 3.5.6. Set <code>snapshot.trust.empty=true</code> in <code>zookeeper.properties</code> before the upgrade to avoid <a href="https://issues.apache.org/jira/browse/ZOOKEEPER-3056">ZOOKEEPER-3056</a>. After the new version starts successfully, it is safe to remove this config. See <a href="https://zookeeper.apache.org/doc/r3.5.6/zookeeperAdmin.html#sc_advancedConfiguration">ZooKeeper advanced configuration</a> for more details.

Reply:
@ijuma Thanks for the review. Addressed the review comment.